tests/utils/test_pytest_parallel_gpu.py

```
@@ -186,6 +201,143 @@ def test_blocked_test_launches_once_occupant_frees():
+    except OSError:
+        return False
+    return True
+
+
+def test_kills_the_whole_process_group_not_just_the_child(tmp_path):
+    # The failure this guards: killing only the pytest child leaves its engine
+    # workers alive holding VRAM, poisoning every test scheduled after it. The
+    # backgrounded grandchild here stands in for those workers -- it survives a
+    # plain proc.kill() and dies only when the whole group is signalled.
+    pid_file = tmp_path / "grandchild.pid"
+    proc = subprocess.Popen(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)